sklearn/neural_network/_stochastic_optimizers.py

```diff
@@ -26,22 +22,25 @@ class BaseOptimizer:
         the current learning rate
     """
 
-    def __init__(self, params, learning_rate_init=0.1):
-        self.params = [param for param in params]
+    def __init__(self, learning_rate_init=0.1):
         self.learning_rate_init = learning_rate_init
         self.learning_rate = float(learning_rate_init)
 
-    def update_params(self, grads):
+    def update_params(self, params, grads):
         """Update parameters with given gradients
 
         Parameters
         ----------
+        params : list, length = len(coefs_) + len(intercepts_)
```

Comment:
Not sure anymore what is the standard there but we could make it better

```suggestion
        params : list of shape (n_coefs_ + intercepts_,)
```

Reply:
I think we went with `list of length`, since list do not have "shapes".